src/atsu/src/InputValidation.ts

```diff
@@ -0,0 +1,446 @@
+        if (before === 136) {
+            if (frequencySpacingEnd.findIndex((entry) => entry === twoDigitFraction) === -1) {
+                return AtsuStatusCodes.EntryOutOfRange;
+            }
+        } else if (frequencySpacingOther.findIndex((entry) => entry === twoDigitFraction) === -1) {
+            return AtsuStatusCodes.EntryOutOfRange;
+        }
+
+        return AtsuStatusCodes.Ok;
+    }
+
+    /**
+     * Validates a value that it is compatible with the FCOM format for altitudes and flight levels
+     * @param {string} value The entered scratchpad altitude
+     * @returns An AtsuStatusCodes-value
+     */
+    public static validateScratchpadAltitude(value: string): AtsuStatusCodes {
+        if (/^((FL)*[0-9]{1,3})$/.test(value)) {
+            let flightlevelStr = '';
+
+            if (value.startsWith('FL')) {
+                flightlevelStr = value.substring(2, value.length);
+            } else {
+                flightlevelStr = value;
+            }
+
+            // contains not only digits
+            if (/(?!^\d+$)^.+$/.test(flightlevelStr)) {
+                return AtsuStatusCodes.FormatError;
+            }
+            const flightlevel = parseInt(flightlevelStr);
+
+            if (flightlevel >= 30 && flightlevel <= 410) {
+                return AtsuStatusCodes.Ok;
+            }
+            return AtsuStatusCodes.EntryOutOfRange;
+        }
+
+        if (InputValidation.FANS === FansMode.FansB) {
+            return InputValidationFansB.validateScratchpadAltitude(value);
+        }
+        return InputValidationFansA.validateScratchpadAltitude(value);
+    }
+
+    /**
+     * Checks if a string fits to the distance definition
+     * @param distance The distance candidate
+     * @returns AtsuStatusCodes.Ok if the format is valid
+     */
+    public static validateScratchpadDistance(distance: string): AtsuStatusCodes {
+        if (/^[0-9]{1,3}(NM|KM)$/.test(distance) || /^[0-9]{1,3}$/.test(distance)) {
+            return AtsuStatusCodes.Ok;
+        }
+        return AtsuStatusCodes.FormatError;
+    }
+
+    /**
+     * Validates a value that it is compatible with the FCOM format for lateral offsets
+     * @param {string} value The entered scratchpad offset
+     * @returns An AtsuStatusCodes-value
+     */
+    public static validateScratchpadOffset(offset: string): AtsuStatusCodes {
+        let nmUnit = true;
+        let distance = 0;
+
+        if (/^[LR][0-9]{1,3}(NM|KM)$/.test(offset) || /^[LR][0-9]{1,3}$/.test(offset)) {
+            // format: DNNNKM, DNNNNM, DNNN
+            distance = parseInt(offset.substring(1, 4));
+            nmUnit = !offset.endsWith('KM');
+        } else if (/^[0-9]{1,3}(NM|KM)[LR]$/.test(offset) || /^[0-9]{1,3}[LR]$/.test(offset)) {
+            // format: NNNKMD, NNNNMD, NNND
+            distance = parseInt(offset.substring(0, 3));
+            nmUnit = !(offset.endsWith('KML') || offset.endsWith('KMR'));
```

Comment:
I'd let the regex do the work here too, rather than error-prone string manipulation. In-fact, for this whole file.